### PR TITLE
Move build script to book

### PR DESCRIPTION
**Pull request by jspada**

### CODE REVIEW

File: kimchi/Cargo.toml [hunk 1]
```diff
@@ -46,15 +45,13 @@ groupmap = { path = "../groupmap", version = "0.1.0" }
 mina-curves = { path = "../curves", version = "0.1.0" }
 o1-utils = { path = "../utils", version = "0.1.0" }
 mina-poseidon = { path = "../poseidon", version = "0.1.0" }
+mina-book = { path = "../book", version = "0.1.0" }
```
Comment (o1-david) [inline kimchi/Cargo.toml:48]: this line defeats the purpose :o you're still going to rely on the dependencies there

File: book/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+description = "Documentation and specifications of the proof systems"
+repository = "https://github.com/o1-labs/proof-systems"
+homepage = "https://o1-labs.github.io/proof-systems/"
+documentation = "https://o1-labs.github.io/proof-systems/rustdoc/"
+readme = "README.md"
+edition = "2021"
+license = "Apache-2.0"
+
+[build-dependencies]
+cargo-spec = { version = "0.5.0" }
```
Comment (o1-david) [inline book/Cargo.toml:13]: you probably want to add this crate to the workspace `Cargo.toml`

File: kimchi/Cargo.toml (hunk 1 above)
Comment (jspada) [reply kimchi/Cargo.toml:48]: Well, our original goal was to make the spec as part of build, so without this it will not build, right?

Comment (jspada) [reply kimchi/Cargo.toml:48]: Ah, I see what you mean.. we add it to top level workspace instead.

Comment (mimoo) [reply kimchi/Cargo.toml]: yeah otherwise this still breaks the wasm build

**Pull request merged**